great_expectations/data_context/util.py

```diff
@@ -93,7 +93,7 @@ def instantiate_class_from_config(config, runtime_config, config_defaults=None):
 
     class_name = config.pop("class_name", None)
     if class_name is None:
-        logger.warning("Instantiating class from config without an explicit class_name is dangerous. Consider adding "
```

Comment:
I agree this should definitely not show up during init, but I believe this should indeed be a warning. It appears that the addition of the CTA config option didn't include a class_name, which is triggering the warning newly. I think this line stays the same, and `templates.py` in data context needs to change from:

```
...
    site_index_builder:
        show_cta_footer: True
```

to

```
...
    site_index_builder:
        class_name: DefaultSiteIndexBuilder
        show_cta_footer: True
```